### PR TITLE
Add support for the ingress-specific sampling value

### PR DESCRIPTION
**Pull request by rl-section**
Prefer it over the annotation value if present.
Add unit tests for all of the above.
Tweak reported sampling pct to actually return a percent.

### CODE REVIEW

File: libbeat/processors/actions/drop_sampling.go [hunk 1]
```diff
@@ -11,38 +11,48 @@ import (
...
 	unannotated *atomic.Int64
 	skipped     *atomic.Int64
 	allowed     *atomic.Int64
 	metrics     *PeriodicSamplingMetrics
 }
 
+// Register this processor plugin
 func init() {
-	fmt.Println("registering drop_sampling")
+	logp.Info("registering drop_sampling!!")
```
Comment (jstangroome) [inline libbeat/processors/actions/drop_sampling.go]: `!!`? 😄

File: libbeat/processors/actions/drop_sampling.go [hunk 2]
```diff
@@ -129,6 +155,107 @@ func (d *dropSampling) findSampling(b *beat.Event) (float64, bool) {
...
+	}
+
+	logJSON := common.MapStr{}
+	err = json.Unmarshal([]byte(log), &logJSON)
+	if err != nil {
+		logp.Debug("drop_sampling", "could not unmarshal %s - %s", logJSON, err)
+		return 0.0, false
+	}
+
+	iface, ok = logJSON["log_sample"]
```
Comment (jstangroome) [inline libbeat/processors/actions/drop_sampling.go:200]: What would be the negatives of JSON unmarshalling and checking for a `log_sample` field on every log line without first checking if the `k8s-app` label was `ingress`? Especially if `log_sample` was a sufficiently unique key that it didn't exist unless it was intended to be consumed here?

**Review by jstangroome — APPROVED** [review]

File: libbeat/processors/actions/drop_sampling.go (hunk 1 above)
Comment (rl-section) [reply libbeat/processors/actions/drop_sampling.go]: Hah, a debug change I missed. Fixed.

File: libbeat/processors/actions/drop_sampling.go (hunk 2 above)
Comment (rl-section) [reply libbeat/processors/actions/drop_sampling.go:200]: It is checked first, see line 172/179 as shown below:

	iface, ok = labels["k8s-app"]
	if !ok {
		logp.Debug("drop_sampling", "couldn't find k8s-app label")
		return 0.0, false
	}

	k8sApp := iface.(string)
	if !ok || k8sApp != "nginx-ingress-lb" {
		logp.Debug("drop_sampling", "not ingress %s", k8sApp)
		return 0.0, false
	}

Comment (jstangroome) [reply libbeat/processors/actions/drop_sampling.go:200]: Sorry, I wasn't clear. I'm hypothesizing that we delete lines 160 to 184 and not check that it is ingress, and just go JSON-decoding and looking for a `log_sample` field in any log of any container and honour it. Would that be terrible, why? And would it still be terrible if the field we looked for was `section_io_log_sample_percent` instead (i.e. a name that would not coincidentally appear in an unrelated log for unrelated reasons).

**Pull request closed without merging**